ambry-rest/src/main/java/com.github.ambry.rest/ConnectionStatsHandler.java

```diff
@@ -53,4 +64,15 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
     openConnections.decrementAndGet();
     super.channelInactive(ctx);
   }
+
+  /**
+   * Listen to the handshake future if this is an SSL connection. Log and update metrics if the handshake failed.
+   * @param ctx the {@link ChannelHandlerContext}.
+   */
+  private void logHandshakeStatus(ChannelHandlerContext ctx) {
```

Comment:
You were the one that wrote this code originally :P 
We typically don't have tests for metrics/logging, besides covering these paths in other tests. I think it might be too cumbersome to test without doing all of the setup that we do in PublicAccessLogHandlerTest.